source/reactor/unit_tests/EventHandlersTest.cpp

```diff
@@ -0,0 +1,92 @@
+#include <sys/socket.h>
+
+#include "EventReaderMock.h"
+#include "EventWriterMock.h"
+#include "reactor/EventLoopImpl.h"
+
+#include "gtest/gtest.h"
+
+using namespace u;
+using namespace u::reactor;
+
+using ::testing::InSequence;
+using ::testing::Invoke;
+using ::testing::StrictMock;
+
+constexpr long test_timeout = 100;
+
+namespace {
+
+TEST(EventHandlersTest, read_write_ShouldSucceed)
+{
+    EventLoopImpl loop;
+    StrictMock<EventReaderMock> reader;
+    StrictMock<EventWriterMock> writer;
+
+    int sv[2];
+    int rc = socketpair(AF_UNIX, SOCK_STREAM, 0, sv);
+    ASSERT_NE(-1, rc);
+
+    {
+        InSequence sequence;
+
+        EXPECT_CALL(writer, handleWrite(sv[0], &loop)).WillOnce(Invoke([](int fd, EventLoop *loop) {
```

Comment:
We don't expect any other methods to be called and in this case, if they were called, it would be an error. This behavior around unexpected calls is "configurable": Google Mock is automatically checking this for us because the mock has been declared as `StrictMock<>`
Please read [The Nice, the Strict, and the Naggy](https://github.com/google/googletest/blob/master/googlemock/docs/cook_book.md#the-nice-the-strict-and-the-naggy-nicestrictnaggy)